crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -434,14 +441,42 @@ where
             Event::ChannelClosed {
                 channel_id,
                 reason,
-                user_channel_id: _,
+                user_channel_id,
             } => {
-                let channel = hex::encode(channel_id);
+                let channel_id = hex::encode(channel_id);
+                let user_channel_id = Uuid::from_u128(user_channel_id).to_string();
                 tracing::info!(
-                    %channel,
+                    %user_channel_id,
+                    %channel_id,
                     ?reason,
                     "\nChannel closed",
                 );
+
+                if let Some(mut channel) = self.storage.get_channel(&user_channel_id)? {
```

Comment:
🤔 We should use `spawn_blocking` (preferred when possible) or `block_in_place` anywhere we access storage in an async context.

For all these cases that have been modified (I'm sure it also applies to some other ones too 😛), consider simply wrapping the whole arm in one of those (if there are no `await`s). Otherwise just wrap the potentially blocking code.